### PR TITLE
new branch to pull bpd with review fixes

### PR DESCRIPTION
**Pull request by MajdT51**

### CODE REVIEW

File: src/test/kotlin/org/eclipse/tractusx/custodian/ApplicationTest.kt [hunk 1]
```diff
@@ -17,49 +17,40 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
-package org.eclipse.tractusx.managedidentitywallets
+package org.eclipse.tractusx.custodian
```
Comment (jpbu) [inline src/test/kotlin/org/eclipse/tractusx/custodian/ApplicationTest.kt:20]: I believe this change of package name is not correct

File: src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/plugins/Scheduler.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+fun Application.configureJobs() {
+    val jdbcUrl = environment.config.property("db.jdbcUrl").getString()
+
+    install(Scheduler) {
+        storageProvider = PostgresStorageProvider(initDatabase(jdbcUrl))
+        threads = 5
+    }
+
+    schedule {
+        recurringJob("bpdm-update", Cron.daily()) {
```
Comment (jpbu) [inline src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/plugins/Scheduler.kt:43]: I think there was a comment in the original PR asking what "daily" means - is there a way to configure this in a more specific way, e.g. to run this nightly at 1am UTC?

**Review by matthiashub-porsche — CHANGES_REQUESTED** [review]

File: .env.docker [hunk 3]
```diff
@@ -14,10 +14,15 @@ CX_AUTH_REDIRECT_URL=http://localhost:8080/callback
 CX_AUTH_ROLE_MAPPINGS=create_wallets:add_wallets,view_wallets:view_wallets,update_wallets:update_wallets,delete_wallets:delete_wallets,view_wallet:view_wallet,update_wallet:update_wallet
 CX_AUTH_RESOURCE_ID="ManagedIdentityWallets"
 
-CX_DATAPOOL_URL=https://catenax-bpdm-dev.germanywestcentral.cloudapp.azure.com
-
 ACAPY_API_ADMIN_URL="http://localhost:11000"
 ACAPY_NETWORK_IDENTIFIER="local:test"
 ACAPY_ADMIN_API_KEY="Hj23iQUsstG!dde"
 
+BPDM_DATAPOOL_URL="https://catenax-bpdm-int.demo.catena-x.net"
```
[the remark this answers is not in the document]
Comment (MajdT51) [reply .env.docker:21]: soll ich sie auch von .env.example entfernen?

File: src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/services/BusinessPartnerDataServiceImpl.kt [hunk 4]
```diff
@@ -85,6 +94,53 @@ class BusinessPartnerDataServiceImpl(private val walletService: WalletService) :
...
 
+    override suspend fun pullDataAndUpdateCatenaXCredentialsAsync() {
+        val listOfBpns = walletService.getAllBpns()
+        var now = LocalDateTime.now()
+        var accessToken = getAccessToken()
+        var expireAt = now.plusSeconds(accessToken.expiresIn.toLong())
+        listOfBpns.forEach { bpn ->
+            try {
+                now = LocalDateTime.now()
+                if (now.isAfter(expireAt)) {
```
[the remark this answers is not in the document]
Comment (jpbu) [reply src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/services/BusinessPartnerDataServiceImpl.kt:105]: True, I missed that in my review - to be safe, catching the 401 is probably the better option, a request might still take 10 sec in extreme cases.

File: src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/Application.kt [hunk 5]
```diff
@@ -90,9 +95,24 @@ fun Application.module(testing: Boolean = false) {
...
+    Services.businessPartnerDataService = businessPartnerDataService
+    Services.walletService = walletService
     configureRouting(walletService)
-    appRoutes(walletService, businessPartnerDataService)
 
+    appRoutes(walletService, businessPartnerDataService)
     configurePersistence()
+
+    if (!testing) {
+        configureJobs()
```
[the remark this answers is not in the document]
Comment (MajdT51) [reply src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/Application.kt:116]: we are not using it. i will remove it for now

File: .env.docker (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthiashub-porsche) [reply .env.docker:21]: @MajdT51 in the example it is fine, as when we use it in bash like `set -a; source .env; set +a` for starting it directly via `./gradlew run` it needs to be quoted

File: src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/routes/BusinessPartnerDataRoutes.kt [hunk 6]
```diff
@@ -37,94 +35,90 @@ import org.eclipse.tractusx.managedidentitywallets.services.BusinessPartnerDataS
 
 fun Route.businessPartnerDataRoutes(businessPartnerDataService: BusinessPartnerDataService) {
 
-    route("/businessPartnerData") {
+    route("/refreshBusinessPartnerData") {
```
[the remark this answers is not in the document]
Comment (matthiashub-porsche) [reply src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/routes/BusinessPartnerDataRoutes.kt:38]: Agree with `businessPartnerDataRefresh` – `businessPartnerDataRefreshCycle` in the url path is more confusing I guess =)

**Comment by matthiashub-porsche** [comment]
Postman collection comment: can you remove the acapy `http://localhost:11000/wallet/did/public?did=KeJZ4sf5YG4RG9DSFVdGDE` call and instead add a POST call to `/api/wallets/BPNL000000000000/public` with following payload

```
{
    "verKey": "<your verkey here>"
}
```

**Pull request merged**